### ISSUE TITLE
tsc error "Cannot find type definition file for '@types'."

### ISSUE THREAD

**Issue by mikaraunio** [pixilab/blocks-script#27]
Strangely, with a clean checkout of the repo, I'm now getting this error when running `tsc` (both with `tsc@4.9.5` and `tsc@^5`):

```
error TS2688: Cannot find type definition file for '@types'.
  The file is in the program because:
    Entry point for implicit type library '@types'
```

Not immediately obvious why this happens, but it does go away if you remove the `@types/suncalc` dep. Or if you add an empty `index.d.ts` to the `node_modules/@types` directory, but clearly that's not a solution...

Since `suncalc.js` is bundled already, the type defs could be bundled too, which would make this disappear?

**Comment by mikaraunio** [comment]
Created PR #28 to bundle type defs.

**Comment by TheWizz** [comment]
Even after merging #28 I still get the same error:

```
error TS2688: Cannot find type definition file for '@types'.
  The file is in the program because:
    Entry point for implicit type library '@types'
```

I was under the impression that you've found a way to get rid of that message. Although it seems harmless, it is annoying – especially for new users. But we've not been able to find the reason for it and how to solve it.

Mike

**Comment by mikaraunio** [comment]
The PR does solve the issue. If you still get the message after uninstalling `@types/suncalc`, there might be a stray empty `@types` directory left behind in `node_modules` which would cause it to be displayed.

If you delete `node_modules` and then rerun `npm install`, do you still get the error then?

**Comment by TheWizz** [comment]
I see. I was compiling the 6.1 branch, which showed the error. However 6.0 now compiles cleanly.

I traced this down to the @types/suncalc dependency added in 6.1, like this:

```
  "dependencies": {
    "reflect-metadata": "^0.1.10",
    "@types/suncalc": "^1.8.1"
  },
```

That shouldn't be needed now that suncalc.d.ts is added manually to lib-archive. And once I removed the @types/suncalc dependency, the compilation error went away.

I can't say I fully understood why (adding the @types/suncalc dependency seemed like the right thing to do). But all seems fine now anyway. So many thanks for solving this conundrum.

Mike
